### ISSUE TITLE
Proposal: Kaia Wallet Tracker Telegram Bot

### ISSUE THREAD

**Issue by chandrapratamar** [kaiachain/kaia-dapp-mono#312]
Description

Kaia Wallet Tracker bot that allows users to quickly and easily track, monitor, and explore wallet activities effortlessly. With simple commands, users can:

- Check the native KAIA balance of a wallet
- View all tokens held in the wallet
- Explore the NFT collections associated with the wallet
- Track wallet addresses for real-time transaction notifications
- Label and organize tracked wallets for easy management

Using the bot is super simple. Just send commands like:
/balance 0x... to see address KAIA balance
/tokens 0x... to list all held tokens
/nfts 0x... to view NFT holdings
/track 0x... [Label] to monitor a wallet
/list to see all tracked wallets
/untrack [address/label] to un-track monitored wallet

**Comment by chandrapratamar** [comment]
Repository : https://github.com/darkvallen/Kaia-Wallet-Tracker-Telegram-Bot
Demo : https://github.com/user-attachments/assets/fe70c2d8-40aa-4857-871b-840310840419
Dorahacks Buidl : https://dorahacks.io/buidl/20709

**Comment by zxstim** [comment]
i think this has very small and limited scope. Kaiascan served this well already.

I am ok with you using Kaiascan and telegram bot but it needs to be something with long term direction.

**Comment by darkvallen** [comment]
> Collaborator

im thinking of adding transaction tracker functionality, notifies you each time a transaction occurs in your address

**Comment by chandrapratamar** [comment]
> i think this has very small and limited scope. Kaiascan served this well already.
> 
> I am ok with you using Kaiascan and telegram bot but it needs to be something with long term direction.

im thinking of adding transaction tracker functionality, notifies you each time a transaction occurs in your address

**Comment by zxstim** [comment]
your wallet does that for you already

**Comment by zxstim** [comment]
https://github.com/kaiachain/kaia-dapp-mono/blob/main/builder-program/ideas.md

check out the idea sheet

**Comment by chandrapratamar** [comment]
> your wallet does that for you already

I mean could be useful not only for personal wallet monitoring but also for tracking other wallets, such as whale wallets, enabling better market insights and on-chain analysis, is it viable?

**Comment by zxstim** [comment]
> tracking other wallets, such as whale wallets, enabling better market insights and on-chain analysis

i am ok with this but this is the submission not what you submitted.

so you are building an open source telegram bot that does this tracking for people. i go in and query, find out what whales are doing (like LPing in what tokens, buying what tokens, selling what tokens, etc.)

**Comment by chandrapratamar** [comment]
> i am ok with this but this is the submission not what you submitted.

Yep, at first I thought this submission would be the base, and after that,i will add the tracking functionality.

> so you are building an open source telegram bot that does this tracking for people. i go in and query, find out what whales are doing (like LPing in what tokens, buying what tokens, selling what tokens, etc.)

Yes, Users can use the command `/track [wallet_address]` to add an address they want to monitor. After that, the bot will send notifications each time the address makes a transaction.

**Comment by chandrapratamar** [comment]
![image](https://github.com/user-attachments/assets/080c389b-e59d-4330-8090-dd815cc61472)

@zxstim is this viable?

**Comment by chandrapratamar** [comment]
![image](https://github.com/user-attachments/assets/12f8d94f-3ea0-4f42-b6a4-51dac7a97c08)

@zxstim, could you check? The repository has also been updated to include the tracking functionality that supports multiple address.

**Comment by chandrapratamar** [comment]
@zxstim, have you checked? Any suggestions or anything else?

**Comment by chandrapratamar** [comment]
@zxstim

**Comment by zxstim** [comment]
what's the bot name on telegram to check?

**Comment by chandrapratamar** [comment]
> what's the bot name on telegram to check?

here's the test bot https://t.me/Kaiatest_Bot

**Comment by chandrapratamar** [comment]
@zxstim 
Have you tested it? If not, I can start the bot up again.

**Comment by chandrapratamar** [comment]
@zxstim

**Comment by chandrapratamar** [comment]
@zxstim

**Comment by chandrapratamar** [comment]
@zxstim bot is up if u want to test

**Comment by chandrapratamar** [comment]
@zxstim please could you check this?

**Comment by chandrapratamar** [comment]
@zxstim

**Comment by zxstim** [comment]
i looked at this and i would have to reject. as discussed 

> i am ok with this but this is the submission not what you submitted.
> so you are building an open source telegram bot that does this tracking for people. i go in and query, find out what whales are doing (like LPing in what tokens, buying what tokens, selling what tokens, etc.)

you are just building a worse version of a block explorer rather than adding values on top

think about scanner that lets me know smart traders, whales, etc.

you have flipside, dune to build system to find out about these activities then use telegram bot as an interface (but probably better on a web interface).
